plugins/permission/src/apis/IdentityPermissionApi.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ConfigApi,
+  DiscoveryApi,
+  IdentityApi,
+} from '@backstage/core-plugin-api';
+import { PermissionApi } from './PermissionApi';
+import {
+  AuthorizeRequest,
+  AuthorizeResponse,
+  PermissionClient,
+} from '@backstage/plugin-permission-common';
+
+/**
+ * The default implementation of the PermissionApi, which simply initializes a {@link PermissionClient} and calls its
+ * authorize method.
+ * @public
+ */
+export class IdentityPermissionApi implements PermissionApi {
+  private readonly permissionClient: PermissionClient;
+
+  constructor(
+    discoveryApi: DiscoveryApi,
+    private readonly identityApi: IdentityApi,
+    configApi: ConfigApi,
```

Comment:
If we require a `PermissionClient`, the frontend plugin will need a dependency on `@backstage/plugin-permission-common` (whereas in our `@backstage/plugin-catalog` example, it currently does not). I don't think we need to make `discoveryApi` or `configApi` `private readonly` because they don't need to be instance variables at all - they're just used to initialize `PermissionClient` in the constructor.

Reply:
It's unfortunate to _create_ things in a constructor though, as an overall pattern. Better to pass in the instance in the constructor, and possibly make a separate `static create(options: { deps... }): IdentityPermissionApi` method if it makes sense to have that too

Reply:
Fair enough! I've changed the constructor to accept a PermissionClient. I don't think we need a static create method for now, but happy to circle back if people feel strongly.

Reply:
Upon integrating this with the `mob/authorization-frontend` branch, it seems that the create method is a clean way of keeping the dependency of `PermissionClient` inside of `IdentityPermissionApi`, so I've implemented that as well. (See [this diff's](https://github.com/backstage/backstage/compare/mob/authorization-framework...mob/authorization-framework-x-permission-frontend) `packages/app-defaults/src/defaults/apis.ts` for usage)